doc/contribute/committer_expectations.rst

```diff
@@ -0,0 +1,287 @@
+get reviewed and merged into the project. The RFC should also define the minimum
+viable implementation.
+
+Changes which require an RFC proposal include:
+
+- Submitting a new feature.
+- Submitting a new API.
+- :ref:`treewide-changes`.
+- Other large changes that can benefit from the RFC proposal process.
+
+Maintainers have the discretion to request committers create an RFC for PRs that
+are too large or complicated.
+
+PR Requirements
+***************
+
+- Each commit in the PR must provide a commit message following the
+  :ref:`commit-guidelines`.
+
+- All files in the PR must comply with :ref:`Licensing
+  Requirements<licensing_requirements>`.
+
+- Follow the Zephyr :ref:`coding_style` and :ref:`coding_guidelines`.
+
+- PRs must pass all CI checks. This is a requirement to merge the PR. Committers
+  may mark a PR as draft and explicitly request reviewers to provide early
+  feedback, even with failing CI checks.
+
+- When breaking a PR into multiple commits, each commit must build cleanly. The
+  CI system does not enforce this policy, so it is the PR author's
+  responsibility to verify.
+
+- When major new functionality is added, tests for the new functionality shall
+  be added to the automated test suite. All API functions should have test cases
+  and there should be tests for the behavior contracts of the API. Maintainers
+  and reviewers have the discretion to determine if the provided tests are
+  sufficient. The examples below demonstrate best practices on how to test APIs
+  effectively.
+
+    - `Kernel timer tests`_ provide better than 85% `test coverage`_ for the
+      kernel timer.
+    - Emulators for off-chip peripherals are an effective way to test driver
+      APIs. The `smart battery sensor tests`_ use the `smart battery emulator`_,
+      providing test coverage for the `fuel gauge API`_ and the `smart battery
+      sensor driver`_
+
+
+- Incompatible changes to APIs must also update the release notes for the
+  next release detailing the change.  APIs marked as experimental are excluded
+  from this requirement. [#api-updates]_
+
+- Changes to APIs must increment the API version number according to the API
+  version rules. [#api-version]_
+
+- PRs must also satisfy all :ref:`merge_criteria` before a member of the release
+  engineering team merges the PR into the zephyr tree.
+
+Maintainers may request committers break up a PR into smaller PRs and may
+request the committer create an :ref:`RFC proposal <rfcs>`.
+
+.. _Kernel timer tests: https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/kernel/timer/timer_behavior
+
+.. _test coverage: https://app.codecov.io/gh/zephyrproject-rtos/zephyr/blob/main/kernel/timer.c
+
+.. _`smart battery sensor tests`: https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/sensor/sbs_gauge/src/test_sbs_gauge.c
+
+.. _`smart battery emulator`: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/sbs_gauge/emul_sbs_gauge.c
```

Comment:
Blocking: please use `:zephyr_file:` to link to files in the tree (grep for examples if unfamiliar). This ensures that release HTML pages point to the corresponding files for that release, unlike direct links, which result in links to files from the future in past release HTML.